classes/class-wc-connect-taxjar-integration.php

```diff
@@ -15,6 +15,131 @@ public function __construct( WC_Connect_API_Client $api_client ) {
 
 	public function init() {
 		// TODO: check if TaxJar plugin is enabled, abort if so
-		// TODO: backup existing tax rates if enabled
+
+		// TODO: check if WCS Taxes are enabled before backup existing tax rates
+		$this->backup_existing_tax_rates();
+	}
+
+	/**
+	 * Exports existing tax rates to a CSV and clears the table.
+	 *
+	 * Ported from TaxJar's plugin.
+	 * See: https://github.com/taxjar/taxjar-woocommerce-plugin/blob/42cd4cd0/taxjar-woocommerce.php#L75
+	 */
+	public function backup_existing_tax_rates() {
+		global $wpdb;
+
+		// Export Tax Rates
+		$rates = $wpdb->get_results( $wpdb->prepare(
+			"SELECT * FROM {$wpdb->prefix}woocommerce_tax_rates
+			ORDER BY tax_rate_order
+			LIMIT %d, %d
+			",
+			0,
+			10000
+		) );
+
+		ob_start();
+		$header =
+			__( 'Country Code', 'woocommerce' ) . ',' .
+			__( 'State Code', 'woocommerce' ) . ',' .
+			__( 'ZIP/Postcode', 'woocommerce' ) . ',' .
+			__( 'City', 'woocommerce' ) . ',' .
+			__( 'Rate %', 'woocommerce' ) . ',' .
+			__( 'Tax Name', 'woocommerce' ) . ',' .
+			__( 'Priority', 'woocommerce' ) . ',' .
+			__( 'Compound', 'woocommerce' ) . ',' .
+			__( 'Shipping', 'woocommerce' ) . ',' .
+			__( 'Tax Class', 'woocommerce' ) . "\n";
+
+		echo $header;
+
+		foreach ( $rates as $rate ) {
+			if ( $rate->tax_rate_country ) {
+				echo esc_attr( $rate->tax_rate_country );
+			} else {
+				echo '*';
+			}
+
+			echo ',';
+
+			if ( $rate->tax_rate_country ) {
+				echo esc_attr( $rate->tax_rate_state );
```

Comment:
I know this isn't your code but... shouldn't this be `if ( $rate->tax_rate_state ) {`?